cpu/nrf5x_common/periph/i2c_nrf52_nrf9160.c

```diff
@@ -331,6 +339,31 @@ int i2c_read_regs(i2c_t dev, uint16_t addr, uint16_t reg,
 
 int i2c_write_bytes(i2c_t dev, uint16_t addr, const void *data, size_t len,
                     uint8_t flags)
+{
+    if ((unsigned int)data >= CPU_RAM_BASE && (unsigned int)data < CPU_RAM_BASE + CPU_RAM_SIZE) {
```

Comment:
This is done a bit simpler for spi: https://github.com/RIOT-OS/RIOT/blob/1073df2ce2804e3290467cba7ca12e36b0d557d5/cpu/nrf5x_common/periph/spi_nrf52_nrf9160.c#L60

Would that work here, too? Is that to relaxed for SPI?

Reply:
... embassy-nrf does this:

```
const SRAM_LOWER: usize = 0x2000_0000;
const SRAM_UPPER: usize = 0x3000_0000;
ptr >= SRAM_LOWER && (ptr + len * core::mem::size_of::<T>()) < SRAM_UPPER

Reply:
The SPI version is probably good enough, we could unify. The embassy version is IMO excessive, as the slice is a single object, and objects don't span memmapping areas.